### ISSUE TITLE
php/switch-to-match wrong transformation

### ISSUE THREAD

**Issue by lucatrev** [laravel-shift/workbench-issues#9]
**Desktop:**
 - OS: Windows
 - OS Version: 10
 - App Version: 0.3.1.0

**Describe the bug**
In specific following scenario tranformation applied is not equivalent to original snippet (tests are failing proving that ✌).
You can notice the tranformed form is missing the orginal default value (`$quote = 0` removed) and the original return (`return $quote`). 
In the expected result is my suggested form given this scenario. 

**To Reproduce**
1. Given the following snippet:
```php
    public function quote()
    {
        $quote = 0;
        switch ($this->comboType) {
            case self::COMBO_OR:
                $quote = $this->quoteForComboOr();
                break;
            case self::COMBO_AND:
                $quote = $this->quoteForComboAnd();
                break;
        }

        return $quote;
    }
```
2. Run 'switch to match' task on it

**Expected result**
```php
    public function quote()
    {
        return match ($this->comboType) {
            self::COMBO_OR => $this->quoteForComboOr(),
            self::COMBO_AND => $this->quoteForComboAnd(),
            default => 0,
        };
    }
```

**Actual result**
```php
    public function quote()
    {
        $quote = match ($this->comboType) {
            self::COMBO_OR => $this->quoteForComboOr(),
            self::COMBO_AND => $this->quoteForComboAnd(),
            default => $quote,
        };
    }
```

Please let me know if you need any further information 👍

PS: shift and its workbench rocks 🤘

**Comment by jasonmccreary** [comment]
@luke83 thanks!

For clarity, is the *actual* result indeed missing the `$quote = 0;` assignment and `return` statement? Or is this task generating:

```php
    public function quote()
    {
        $quote = 0;
        $quote = match ($this->comboType) {
            self::COMBO_OR => $this->quoteForComboOr(),
            self::COMBO_AND => $this->quoteForComboAnd(),
            default => $quote,
        };

        return $quote;
    }
```

**Comment by lucatrev** [comment]
Yep, @jasonmccreary, i confirm the _actual_ result is missing `$quote = 0;`, I would never have noticed without failing tests 😅 

Of course another equivalent form could be:

```php
// expected version #2
public function quote()
    {
        $quote = 0;
        return match ($this->comboType) {
            self::COMBO_OR => $this->quoteForComboOr(),
            self::COMBO_AND => $this->quoteForComboAnd(),
            default => $quote,
        };
    }
```

if it is easier to implement 😎

**Comment by jasonmccreary** [comment]
Thanks for the clarification. I'll get @jessarcher involved on this one and we'll get it patched up in a future release.

Nice work having tests!

**Comment by lucatrev** [comment]
Initial assignment and return statement are gone, after task, but only in this scenario, i have other 5 switches handled correctly 💪

![image](https://user-images.githubusercontent.com/77144/124791143-13f0f880-df4c-11eb-8854-8550f07689fd.png)

tests, "this is the way", as Mando would say!

**Comment by jessarcher** [comment]
This issue has now been fixed in Rector. Just waiting for them to release an update containing the fix.

The output won't be exactly as you've mentioned because the early return is outside the scope of that rector task, but there is a separate task that should be able to make this change as well.

**Comment by jasonmccreary** [comment]
@luke83 this should be resolved in the latest release (0.4.6). Let us know.
